src/pages/api/stripe/webhook.ts

```diff
@@ -17,8 +29,43 @@ const stripeWebhookHandler = async (
     try {
       event = stripe.webhooks.constructEvent(buf, sig as string, webhookSecret)
 
-      if (event.type === 'checkout.session.completed') {
-        console.log('stripe webhook handler', {event})
+      // Also handle:
+      // - 'customer.subscription.updated'
+      // - 'customer.subscription.deleted'
+      if (event.type === 'customer.subscription.created') {
+        let stripeCustomerId = event.data.object.customer
+
+        let stripeCustomer = await stripe.customers.retrieve(stripeCustomerId)
+
+        // TODO: not sure how to address this type error with Stripe's Customer
+        // type
+        let stripeEmail = stripeCustomer.email
+
+        // we want to get the `co_************` id from CustomerIO which we can
+        // cross-reference with MixPanel to uniquely identify the user.
+        let cioCustomer = await cioAxios
+          .get(`/customers`, {
+            params: {email: stripeEmail},
+            headers: {
+              Authorization: `Bearer ${process.env.CUSTOMER_IO_APPLICATION_API_KEY}`,
+            },
+          })
+          .then(({data}: {data: any}) => data.results[0])
+          .catch((error: any) => {
+            console.error(error)
+            return {}
+          })
+
+        // if running sale we would want to check for that but the regular case it's pro or PPP
+        // TODO: need to differentiate between other types of discounts (e.g. sales)
+        let subscriptionType = !event.data.object.discount ? 'pro' : 'PPP'
+        let interval = event.data.object.plan.interval
+
+        mixpanel.people.set(cioCustomer.id, {
+          pro: true,
+          type: subscriptionType,
+          interval,
+        })
```

Comment:
Currently we are just setting profile property values. I'm fuzzy on how mixpanel will work for building reports if we will also have to send events to properly track the sign up rate